File: app/src/main/java/com/omarea/common/ui/BlurBottomBarLayout.java
package com.omarea.common.ui;

import android.content.Context;
import android.util.AttributeSet;

public class BlurBottomBarLayout extends BlurViewLinearLayout {
    public BlurBottomBarLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        // Giao diện "đảo nổi": giữ bo góc mặc định thay vì tắt về 0, để thanh dưới trông
        // như một khối nổi tách biệt khỏi cạnh màn hình (kết hợp với margin ở layout XML).
        // Không override drawStroke() nữa: dùng viền bo tròn đầy đủ 4 cạnh của lớp cha
        // thay vì chỉ vẽ 1 đường kẻ ở cạnh trên.
    }
}

File: app/src/main/kotlin/com/omarea/common/ui/BlurBottomBarLayout.kt
package com.omarea.common.ui

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet

class BlurBottomBarLayout(context: Context, attrs: AttributeSet?) : BlurViewLinearLayout(context, attrs) {
    init {
        // Thanh dưới thường trải dài hết màn hình nên tắt bo góc để khớp với cạnh máy
        this.engine.cornerRadius = 0f
    }

    override fun drawStroke(canvas: Canvas) {
        // Lấy Paint tĩnh từ BlurEngine để đảm bảo màu sắc thay đổi theo Dark/Light Mode
        val paint = BlurEngine.getStrokePaint(context)
        val strokeWidth = paint.strokeWidth

        // GIẢI PHÁP: Vẽ đường kẻ ở cạnh trên cùng (Top Edge) của Bottom Bar
        // Thụt xuống một nửa độ dày viền (strokeWidth / 2f) để nét vẽ nằm trọn bên trong View
        val y = strokeWidth / 2f

        canvas.drawLine(0f, y, width.toFloat(), y, paint)
    }
}
